nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CountText.java

```diff
@@ -213,10 +221,10 @@ public void onTrigger(ProcessContext context, ProcessSession processSession) thr
         }
         AtomicBoolean error = new AtomicBoolean();
 
-        lineCount = 0;
-        lineNonEmptyCount = 0;
-        wordCount = 0;
-        characterCount = 0;
+        final AtomicInteger lineCount = new AtomicInteger(0);
+        final AtomicInteger lineNonEmptyCount = new AtomicInteger(0);
+        final AtomicInteger wordCount = new AtomicInteger(0);
+        final AtomicInteger characterCount = new AtomicInteger(0);
```

Comment:
Since we've moved these variables from volatile instance variables to local variables inside this onTrigger method, I believe we don't have to use AtomicIntegers, primitive ints should suffice.

Reply:
Hi Koji,thanks for the review!! 
I don't think I'm very clear, I'll try to explain better why I don't use primitive types.
Lambda expressions (as well as anonymous classes) in Java can only access to the final (or effectively final) variables of the enclosing scope.
For example, consider the following example:

```
void fn() {
    int myVar = 42;
    Supplier<Integer> lambdaFun = () -> myVar; // error
    myVar++;
    System.out.println(lambdaFun.get());
}
```
This doesn't compile since the incrementation of myVar prevents it from being effectively final.

In my original code, before changing it to Atomic integer, I use a array since java does not have this problem when you use an array, because arrays are reference objects. Java can capture array reference that never changes, and use that non-changing reference to mutate the object. Array itself provides the necessary level of indirection, because Java arrays are mutable.

If I declare them local within the lambda expression, cause the code block that calls putAllAttributes to fail since they will not be accessible.Check other implementations of this type in nifi, for example the ForkRecord processor and see that it uses atomicInteger for this type of references.

Reply:
To change this implementation by local primitive variables to the lambda expression, you should refactor a lot of code, for example the way in which the flowfile attributes are being configured when transferring to the success relationship.

Reply:
Thanks for the explanation. I was looking at the only changed code and didn't pay attention to the lambda part. That makes sense.